src/controllers/campaign/delete.ts

```diff
@@ -0,0 +1,27 @@
+import { AppResponse } from '@/common/utils';
+import AppError from '@/common/utils/appError';
+import { catchAsync } from '@/middlewares';
+import { Request, Response } from 'express';
+import { campaignModel as campaign } from '@/models/campaignModel';
+import { DateTime } from 'luxon';
+
+const deleteCampaign = catchAsync(async (req: Request, res: Response) => {
+	const { user } = req;
+
+	if (!user) {
```

Comment:
this check is not required as this is a protected route and there will always be a user